src/main/kotlin/org/jetbrains/numkt/NativeLibUtils.kt

```diff
@@ -16,141 +16,102 @@
-        append("META-INF/Native/")
-        when (osName) {
-            OSType.MACOS -> append("macos")
-            OSType.LINUX -> append("linux")
-            OSType.WINDOWS -> append("win")
-            else -> {
+    init {
+        val dirScr = File("buildScr/python")
+        if (dirScr.exists()) {
+            tmpDir = dirScr
+        } else {
+            tmpDir = Files.createTempDirectory("nativeKtNumPy").toFile().apply { deleteOnExit() }
+
+            //extract py script and native lib from jar
+            mapOf(
+                "/META-INF/pythonScript/$namePythonScript.py" to "$namePythonScript.py",
+                "/META-INF/Native/$nameNativeLib" to nameNativeLib
+            ).forEach {
+                extractFileFromJar(it.key, it.value)
             }
         }
-        append("/objs")
-    }
-
-    if (Interpreter::class.java.getResource("/$nativeLibraryPath") == null) {
-        throw Exception("Error loading native library: $libName, missing objs from $nativeLibraryPath")
     }
 
-    // create temp dir for uploading objs there
-    // and delete it upon completion work vm
-    val tempNativeFolder = Files.createTempDirectory("nativeKtnumpy").toFile()
-    tempNativeFolder.deleteOnExit()
+    fun loadLibraries() {
 
-    // extract objs, script and install lib to PATH
-    extractAndCompl(nativeLibraryPath, tempNativeFolder)
-
-    // load library from PATH
-    System.load(libDest.absolutePath)
-}
-
-
-fun extractAndCompl(nativeLibraryPath: String, tempNativeFolder: File) {
-
-    // extract objs from jar
-    // get jar
-    val jarFile = File(Interpreter::class.java.protectionDomain.codeSource.location.path.replace("%20", " "))
-
-    var pythonTmpScript: File? = null
-    val namePythonScript = "buildKtNumPy.py"
-
-    if (jarFile.isFile) {
-        val jar = JarFile(jarFile)
-        val entries = jar.entries()
-        while (entries.hasMoreElements()) {
-            val element = entries.nextElement()
-
-            if (element.name.contains(namePythonScript)) {
-                if (element.name.substringAfterLast("/") == "")
-                    continue
-                pythonTmpScript = File(tempNativeFolder, namePythonScript)
-                pythonTmpScript.deleteOnExit()
-                Interpreter::class.java.getResourceAsStream("/${element.name}")
-                    .use { `in` -> Files.copy(`in`, pythonTmpScript.toPath()) }
-                continue
-            }
+        //load python lib
+        if (!File(pythonConf.pythonLibPath).exists()) {
+            throw FileNotFoundException(pythonConf.pythonLibPath)
+        }
+        System.load(pythonConf.pythonLibPath)
+
+        val exceptionMessage = StringBuilder()
+
+        //load ktnumpy from java.library.path
+        try {
+            System.loadLibrary(baseNameNativeLib)
+            return
+        } catch (e: UnsatisfiedLinkError) {
+            exceptionMessage
+                .append(e.message)
+                .append("\n")
```

Comment:
What if we catch the exception here, and don't catch it in the block below? It seems that in such case execution will go further. It seems that we should check the error presence after all try-catch blocks, and then throw

Reply:
Initially, loading the library from `java.library.path`, if successful, return.
If unsuccessful, then save error message and load in the absolute path.